### DISCUSSION TITLE
Resize and center windows?

### DISCUSSION THREAD

**Discussion by ismk** [chrenn/swish-dl#1] [Q&A]
Hello!
First of all thank you making Swish! I love it and have been using it for sometime now. With the addition of keyboard shortcuts it has replaced all the alternate window snapping apps like Rectangle and Magnet for me.

One question, is there a way I can resize and center a window? 
The only way I found is first making the window quater size and then double clicking it to center it, is there a fast way?

Thanks!

**Comment by chrenn** [comment]
The center gesture uses the unsnapped size of the window, so I'm not sure how snapping the window to a quarter before would help? Could you elaborate?

**Reply by ismk** [reply]
yes you are right it does not. So my quesiton is: 
Is there a way (or hopefully in the future) where we can resize and center a window on the screen?

**Reply by chrenn** [reply]
Still not sure what you mean. The center gesture pretty much does exactly that. Can you be more specific?

**Reply by ismk** [reply]
Sorry I'll try to be more specific, other window managers have an option of resizing the current window and centering it. Example Moom goes by grid based layout that you can specify the amount of `cells` the window occupies when centering the window. Example: 
<img width="499" alt="Screen Shot 2021-05-06 at 12 12 01" src="https://user-images.githubusercontent.com/7370308/117338604-62a6e700-ae64-11eb-95a8-cf1d45b161d8.png">

Now I know swish is not grid based but I was wondering if there was a way to absolute center a window with relative width and height depending on the screen size like rectangle does here:

![](https://user-images.githubusercontent.com/13651296/101402672-57ab5300-38d4-11eb-9e8c-6a3147d26711.png)

What I believe swish is doing is looking at the last value for the width and height of the window and reverting to that when we use the centering double tap gesture. For example if I resize the window to almost full screen size it will center that without resizing it.



https://user-images.githubusercontent.com/7370308/117341566-e9a98e80-ae67-11eb-9239-e852e2cb5a43.mp4



I am wondering if there is a way to resize and center a window consistently to a size no matter the previous size of that window had.

**Reply by chrenn** [reply]
So like the 'Almost Maximize' setting in Rectangle? Swish's 'Center' gesture unsnaps the window (if previously snapped) before centering it.

**Reply by ismk** [reply]
No I meant the center setting in Rectangle, regardless of its previous size, the snapping or unsnapping the window always resizes and centers the window on the screen with a fixed size.

Like you said the 'Center' gesture unsnaps the window first then centers (when this unsnaps, if the window is quite large, it centers it without resizing it) 
What I am asking can we have it so that when we use the 'Center' gesture it does not  unsnap and return the window to its original size but snap and resize the window to center. Does that make sense? 

Please let me know if you want me to demonstrate this somehow.

**Reply by chrenn** [reply]
So you want to be able to center a snapped window without unsnapping it first?

**Reply by ismk** [reply]
Yes! 
And I would like to correct myself, I downloaded Rectangle and its 'Almost Maximize' setting is the functionality I am talking about. Sorry the different terms all the window managers use are confusing.

**Reply by chrenn** [reply]
There will be a switch for centering/unsnapping in the coming update. No promises on 'Almost Maximizing' tho.

**Reply by ismk** [reply]
Thats awesome!! 🍾 
Thank you
